scripts/release

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env ruby
+
+require "date"
+require "optparse"
+
+def run(command, abort_on_fail: true)
+  puts "> #{command}" if $VERBOSE
+
+  system(command, exception: abort_on_fail)
+rescue => e
+  raise "#{command.inspect} failed: #{e}"
+end
+
+channel = ENV.fetch("FOREM_RELEASE_CHANNEL", "stable")
+date = ENV.fetch("FOREM_RELEASE_DATE", Date.today.to_s).gsub(/-/, "")
+hotfix = ENV.fetch("FOREM_RELEASE_HOTFIX", 0)
+remote = ENV.fetch("FOREM_RELEASE_REMOTE", "origin")
+source = ENV.fetch("FOREM_STAGING_FROM", "#{channel}-next")
+
+OptionParser.new ARGV.dup do |options|
+  options.banner = <<~USAGE
+    Usage: #$0 [options]
+
+    Releases are named CHANNEL.DATE.HOTFIX_COUNT. Example: #{channel}.#{date}.#{hotfix}
+
+  USAGE
+
+  options.on "-c", "--channel CHANNEL", "Specify the release channel (defaults to #{channel.inspect})" do |release_channel|
+    channel = release_channel
+  end
+
+  options.on "-d", "--date DATE", "Specify the release date (defaults to today)" do |release_date|
+    date = release_date.gsub(/-/, "")
+  end
+
+  options.on "-f", "--from BRANCH/COMMIT/TAG", "Specify the source branch for this release (defaults to #{source.inspect})" do |from_branch|
+    source = from_branch
+  end
+
+  options.on "-h", "--hotfix HOTFIX", "Specify the hotfix count for this release branch (defaults to #{hotfix.inspect})" do |hotfix_opt|
+    hotfix = hotfix_opt
+  end
+
+  options.on "-r", "--remote REMOTE", "Git remote to push to (defaults to #{remote.inspect})" do |git_remote|
+    remote = git_remote
+  end
+
+  options.on "-v", "--verbose", "Enable verbose mode" do
+    $VERBOSE = true
+  end
+end.parse!
+
+original_branch = `git branch --show-current`.strip
+
+begin
+  tag = "#{channel}.#{date}.#{hotfix}"
+  puts "Building #{tag}..."
+
+  [source, channel].each do |branch|
+    run "git checkout --quiet #{branch}"
+    run "git pull --quiet --ff-only #{remote} #{branch}"
+    puts "Pulled latest #{branch.inspect}"
+  end
+
+  run "git merge --quiet #{source}"
+  puts "Merged #{source} into #{channel}"
+
+  begin
+    run "git tag #{tag}"
+    puts "Tagged #{tag}"
+  rescue
```

Comment:
Hm, generally Rubocop complains when we don't specify an error class, I'm surprised it didn't here (and in `run` above).

Reply:
@jgaskins I ran Rubocop against this file and it seems that for some reason your pre-commit hook didn't do that, so maybe do it manually instead:

```
1 file inspected, 19 offenses detected, 15 offenses corrected, 1 more offense can be corrected with `rubocop -A`
```

Reply:
Looks like there's [no linter rule for these files](https://github.com/forem/forem/blob/0d58de830ba98fa8bfbfd8b88491e3a4e2353c7b/package.json#L36-L77), so I just added one in 691dc3286.

Reply:
Ah, that explains it, thanks Jamie!